### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by anthonynmh**
AddCommand implementation
1. Add elaboration for Add command implementation
2. Update activity diagram for Add command

RecurCommand implementation
1. Add elaboration for Recur command implementation
2. Update activity diagram for Recur command - include rake for _event clash check_
3. Add diagram for _event clash check_

EditCommand implementation
1. Add elaboration for Edit command implementation

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -175,22 +175,125 @@ Classes used by multiple components are in the `ezschedule.commons` package.
...
+-- user input --  
+Step 1. User executes add command with correct and valid arguments.
+
+-- `SchedulerParser` --  
+Step 2. Returns new `EditCommandParser`.
+
+-- `EditCommandParser` --   
+Step 3. Verify that all argument prefixes are present.  
+Step 4. Verify that all argument format is valid.    
+Step 5. Returns new `EditCommand`.
```
Comment (JrmCkh) [inline docs/DeveloperGuide.md:285]: change to:
Step 3. Verify that at least one of the argument prefixes are present.  
Step 4. Verify that provided arguments are valid.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -175,22 +175,125 @@ Classes used by multiple components are in the `ezschedule.commons` package.
...
+- `EditCommandParser.java` - For parsing the arguments to `EditCommand.java`.
+- `EditCommand.java` - For execution.
+
+The following exceptions may be thrown during this process, namely:
+- ParseException for missing arguments
+- ParseException for invalid arguments
+- CommandException for index out of range
+- CommandException for identical events
+- CommandException for events with clashing time
+
```
Comment (JrmCkh) [inline docs/DeveloperGuide.md:273]: Add the following as well:
InvalidDateException for correct syntax but invalid (do not exist) dates

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -175,22 +175,125 @@ Classes used by multiple components are in the `ezschedule.commons` package.
...
+Given below is an example usage scenario of how the _Edit_ command executes.
+
+-- user input --  
+Step 1. User executes add command with correct and valid arguments.
+
```
Comment (JrmCkh) [inline docs/DeveloperGuide.md:278]: edit command, not add

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (anthonynmh) [reply docs/DeveloperGuide.md:275]: Added line 271

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (anthonynmh) [reply docs/DeveloperGuide.md:280]: updated

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (anthonynmh) [reply docs/DeveloperGuide.md:287]: updated

**Review by JrmCkh — APPROVED** [review]
LGTM

**Pull request merged**